File: Algorithm/programmers/Level2/N개의 최소 공배수.py
# 효율성 x
def solution(arr):
    answer = 0
    total = 1
    for i in arr:
        total *= i

    for i in range(max(arr), total + 1):
        count = 0
        for j in arr:
            if i % j == 0:
                count += 1

        if count == len(arr):
            answer = i
            break

    return answer

# 효율성 O
import math

def solution(arr):
    answer = arr[0]

    for num in arr:
        answer = num * answer // math.gcd(num, answer)

    return answer

File: Algorithm/programmers/Level2/N개의 최소 공배수.js
function solution(arr) {
  let getGCD = (num1, num2) => {
    let gcd = 1;

    for (let i = 2; i <= Math.min(num1, num2); i++) {
      if (num1 % i === 0 && num2 % i === 0) {
        gcd = i;
      }
    }

    return gcd;
  };

  let answer = arr[0];

  for (let i = 1; i < arr.length; i++) {
    answer = (answer * arr[i]) / getGCD(answer, arr[i]);
  }

  return answer;
}
